pkg/skaffold/kubernetes/manifest/images.go

```diff
@@ -134,3 +142,26 @@ func (r *imageReplacer) Check() {
 		}
 	}
 }
+
+// imageSelector represents a strategy for matching the container `image` defined in a kubernetes manifest with the correct skaffold artifact.
+type imageSelector func(tagsByImageName map[string]string, image *docker.ImageReference) (imageName, tag string, valid bool)
+
+func selectLocalManifestImages(tagsByImageName map[string]string, image *docker.ImageReference) (string, string, bool) {
+	// Leave images referenced by digest as they are
+	if image.Digest != "" {
+		return "", "", false
+	}
+	// local manifest mentions artifact `imageName` directly, so `imageName` is parsed into `image.BaseName`
+	tag, present := tagsByImageName[image.BaseName]
+	return image.BaseName, tag, present
+}
+
+func selectRemoteManifestImages(tagsByImageName map[string]string, image *docker.ImageReference) (string, string, bool) {
+	// if manifest mentions `imageName` directly then `imageName` is parsed into `image.BaseName`
+	if tag, present := tagsByImageName[image.BaseName]; present {
+		return image.BaseName, tag, present
+	}
+	// if manifest mentions image with repository then `imageName` is parsed into `image.Name`
+	tag, present := tagsByImageName[image.Name]
+	return image.Name, tag, present
```

Comment:
if I'm reading this correctly, this block is to account for the existing locally-referenced images, right?

if so, it seems that this block will always be used in selecting the images to replace. it might be a little easier to read to remove this part from the `selectRemoteManifestImages()` function, and then alter the `if len(k.remoteManifests) > 0 ... else` block to just be an `if len(k.remoteManifests) > 0 ...`.

```golang
if len(k.RemoteManifests) > 0 {
		manifests, err = manifests.ReplaceRemoteManifestImages(ctx, builds)
}
manifests, err = manifests.ReplaceImages(ctx, builds)
```

Reply:
made the separation between local and remote manfests cleaner. `ReplaceRemoteManifestImages` still needs to account for images referenced directly similar to local manifests.